### PR TITLE
 Don't show "no guilds" texts while loading

### LINKED ISSUE #10662: My Guilds page shows "You aren't member of any Guilds" message while loading
### Description
The [My Guilds](https://habitica.com/groups/myGuilds) page displays a "You aren't member of any Guilds" message while it is loading my guilds from the server, which is kind of confusing (especially if the loading takes longer than a second for some reason). 

I think it would be better to display either nothing or similar "Loading..." text as the Discover Guilds page while the network request is pending, and only show "You aren't member of any Guilds" message after the request has finished and we actually know the user is not a member of any guilds.

I can make a PR for this if it's OK.

### PR DESCRIPTION
**Pull request by cvuorinen**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #10662

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Previously My Guilds page displayed the "You aren't member of any Guilds" message (and My Challenges page displayed "You don't have any challenges" message) while it was still loading them from the API and also when filtering so that there are no matches.

This PR adds loading indicator (just text "Loading..." as used in Discover Guilds and Discover Challenges pages) that is displayed while the API request is pending and only shows "no guilds" or "no challenges" message after loading is finished. Also, when filtering is used and there are no matches (but user has guilds/challenges), then nothing is shown (as in Discover Guilds and Discover Challenges pages). 

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: fa756acf-9127-4a3c-8dac-8ff627d30073

### CODE REVIEW

**Comment by negue** [comment]
Changes looks good 👍 , but could you change the title color also back to `$gray-200`

**Comment by cvuorinen** [comment]
Oh, forgot to mention in the PR description that I unified the styling of the "no guilds" message to match "no challenges" message. So that's why I changed the color.

Do you think that is something worth pursuing? So if I change it back to 200, should I change "no challenges" text also?

**Comment by paglias** [comment]
@cvuorinen yeah, can you change both the no guilds and no challenges text to gray-200? thanks

Also could you post a screenshot showing the changes

**Comment by cvuorinen** [comment]
I updated the color. And here are the screenshots:

**My Guild page before changes:**

screenshot from 2018-09-10 19-25-38

**My Challenges page before changes:**

screenshot from 2018-09-10 19-25-52

**My Guilds page after changes:**

screenshot from 2018-09-10 19-22-51

**My Challenges page after changes:**

screenshot from 2018-09-10 19-23-02

This PR was supposed to be mostly about the functionality (not showing that message while loading/filtering) but I thought it would be good to unify these two views a bit at the same time. If the old one is better, I can revert the style changes and just leave the functionality.

**Comment by negue** [comment]
ping @Tressley

**Comment by Tressley** [comment]
Just need to lighten the megaphone icon for Challenges (#C3C0C7) and the rest LGTM. 👍

**Comment by cvuorinen** [comment]
Not sure if you guys got notification about my commit or if I'm supposed to comment here as well, but I fixed the icon color already after the last comment.

File: website/client/components/challenges/myChallenges.vue [hunk 1]
```diff
@@ -16,7 +17,7 @@
           span(v-once) {{$t('createChallenge')}}
 
     .row
-      .no-challenges.text-center.col-md-6.offset-3(v-if='filteredChallenges.length === 0')
```
Comment (paglias) [inline website/client/components/challenges/myChallenges.vue:19]: why did you change this from checking `filteredChallenges.length ` to `challenges.length `?

Comment (cvuorinen) [reply website/client/components/challenges/myChallenges.vue:19]: So that it does not say "you don't have any challenges" when in fact you do have, but they just don't match with the filters you selected. I tried to explain it in the PR description.

Comment (paglias) [reply website/client/components/challenges/myChallenges.vue:19]: that makes sens, thanks @cvuorinen

Comment (paglias) [reply website/client/components/challenges/myChallenges.vue:19]: But this means that if you're filtering the challenges and no result matches the filter / search query we're going to show just an empty page? @Tressley there's a design for this?

Comment (cvuorinen) [reply website/client/components/challenges/myChallenges.vue:19]: That's true, but as I said in the description it is already like that in Discover Guilds page and Discover Challenges pages. IMO, that is still better than showing false information. 

But of course best option is to have a different message for the case when filters don't match (but that should probably be a separate PR).

Comment (Tressley) [reply website/client/components/challenges/myChallenges.vue:19]: The design would look similar to the empty state. The messaging would need to be updated to something along the lines of "We couldn't find any Challenges that match."

Comment (cvuorinen) [reply website/client/components/challenges/myChallenges.vue:19]: So should I go ahead and implement something based on that or what is the next course of action?

Or would it be ok to merge this PR first and then make a new one for that? I can work on it if that's okay with everyone, but would like to get this one out of the way since it's been hanging quite long already.

Comment (paglias) [reply website/client/components/challenges/myChallenges.vue:19]: I think this one is good to go! For the other issue let's make a new PR. I'll make sure to finally review this before we merge the next badge of PRs

**Comment by paglias** [comment]
Sorry for the delay @cvuorinen and thanks for the PR! I've notated this towards your 3rd tier
